tests/unit/test_algebraic_expression.cpp

```diff
@@ -654,81 +665,97 @@ TEST_F(AlgebraicExpressionTest, Exp_OP_MUL) {
 
 TEST_F(AlgebraicExpressionTest, Exp_OP_ADD_Transpose) {
 	// Exp = A + Transpose(A)
-	GrB_Matrix A;
 	GrB_Matrix B;
 	GrB_Matrix res;
-
-	// A
-	// 1 1
-	// 0 0
-	GrB_Matrix_new(&A, GrB_BOOL, 2, 2);
-	GrB_Matrix_setElement_BOOL(A, true, 0, 0);
-	GrB_Matrix_setElement_BOOL(A, true, 0, 1);
-
-	// B
-	// 1 1
-	// 1 0
-	GrB_Matrix_new(&B, GrB_BOOL, 2, 2);
-	GrB_Matrix_setElement_BOOL(B, true, 0, 0);
-	GrB_Matrix_setElement_BOOL(B, true, 0, 1);
-	GrB_Matrix_setElement_BOOL(B, true, 1, 0);
-
+	/* We must use matrices that we've added to the graph, or else
+	 * the transpose optimization will erroneously use the adjacency matrix
+	 * (since there is no schema associated with the operands).
+
+	 * The 'visit' matrix represented by V has the form:
+	 * 0 0 1 1
+	 * 0 0 1 0
+	 * 0 0 0 0
+	 * 0 0 0 0
+	 *
+	 * Its transpose, TV, has the form:
+	 * 0 0 0 0
+	 * 0 0 0 0
+	 * 1 1 0 0
+	 * 1 0 0 0
+	 *
+	 * The expected result of the addition is:
+	 * 0 0 1 1
+	 * 0 0 1 0
+	 * 1 1 0 0
+	 * 1 0 0 0
+	 */
+	GrB_Matrix_new(&B, GrB_BOOL, 4, 4);
+	GrB_Matrix_setElement_BOOL(B, true, 0, 2);
+	GrB_Matrix_setElement_BOOL(B, true, 0, 3);
+	GrB_Matrix_setElement_BOOL(B, true, 1, 2);
+	GrB_Matrix_setElement_BOOL(B, true, 2, 0);
+	GrB_Matrix_setElement_BOOL(B, true, 2, 1);
+	GrB_Matrix_setElement_BOOL(B, true, 3, 0);
 	// Matrix used for intermidate computations of AlgebraicExpression_Eval
 	// but also contains the result of expression evaluation.
-	GrB_Matrix_new(&res, GrB_BOOL, 2, 2);
-
-	// A + Transpose(A)
-	rax *matrices = raxNew();
-	raxInsert(matrices, (unsigned char *)"A", strlen("A"), A, NULL);
-	AlgebraicExpression *exp = AlgebraicExpression_FromString("A+T(A)", matrices);
+	GrB_Matrix_new(&res, GrB_BOOL, 4, 4);
+	AlgebraicExpression *exp = AlgebraicExpression_FromString("V+tV", _matrices);
```

Comment:
We cannot, since `_AlgebraicExpression_ApplyTranspose` makes the reasonable assumption that the operands it's working on already exist. Any operand constructed by `AlgebraicExpression_FromString` has no label/type, and as such falls into the `else` block that fetches the overall transposed adjacency matrix:
https://github.com/RedisGraph/RedisGraph/pull/877/files#diff-4fea05ed5f868356d3e1c16637f167eaR441-R446

As such, the only safe test I could come up with was to use the operands we've also actually added to the graph.